### PR TITLE
Implement vitepress template for `ts-utils` docs

### PR DESCRIPTION
**Pull request by jajibhee**

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -4,34 +4,32 @@ specifiers:
   '@algolia/client-search': ^4.15.0
```
Comment (robertkibet) [inline pnpm-lock.yaml:4]: @alexiglesias93 do we need this plugin?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -4,34 +4,32 @@ specifiers:
...
   '@changesets/changelog-git': ^0.1.14
   '@changesets/cli': ^2.26.0
-  '@finsweet/docs-theme': ^0.1.2
```
Comment (robertkibet) [inline pnpm-lock.yaml]: I will add this after merging the docs-theme PR https://github.com/finsweet/docs-theme/pull/17

File: pnpm-lock.yaml (hunk 1 above)
Comment (robertkibet) [inline pnpm-lock.yaml:4]: @alexiglesias93 do we need this for our usecase?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -4,34 +4,32 @@ specifiers:
...
   typescript: ^4.9.5
-  vitepress: ^1.0.0-alpha.49
+  vitepress: 1.0.0-alpha.50
   vue: ^3.2.47
 
 devDependencies:
   '@algolia/client-search': 4.15.0
   '@changesets/changelog-git': 0.1.14
   '@changesets/cli': 2.26.0
-  '@finsweet/docs-theme': link:node_modules/@finsweet/docs-theme
```
Comment (robertkibet) [inline pnpm-lock.yaml]: I will add this after this is merged https://github.com/finsweet/docs-theme/pull/17

File: pnpm-lock.yaml (hunk 1 above)
Comment (alexiglesias93) [reply pnpm-lock.yaml:4]: nope, please uninstall it

**Comment by robertkibet** [comment]
Will address a couple of items and feature additions as highlighted in slack before making it ready for review.

File: pnpm-lock.yaml (hunk 3 above)
Comment (robertkibet) [reply pnpm-lock.yaml]: updated: https://github.com/finsweet/ts-utils/pull/24/commits/a435a84e0f25283b1282f50904f7e0cc95079ac8

File: pnpm-lock.yaml (hunk 1 above)
Comment (robertkibet) [reply pnpm-lock.yaml]: removed: https://github.com/finsweet/ts-utils/pull/24/commits/a435a84e0f25283b1282f50904f7e0cc95079ac8

**Pull request merged**